pkg/util/tls/tls.go

```diff
@@ -41,6 +50,13 @@ func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
 		caCertPool = x509.NewCertPool()
 		caCertPool.AppendCertsFromPEM(caCert)
 		if len(clientCert.Certificate) > 0 && caCertPool != nil {
+			prometheus.MustRegister(tlsCertNotAfterTimestamp)
```

Comment:
Please use new `promauto` package and give it a registrerer.

Reply:
It's not even correct register it here, because if the `GetTLSConfig()` is called twice, it will panic because of a double registration. You can just use `promauto.NewGaugeVec()` above, even if we should stop using the global prometheus registerer, but if you really want this metric then using the global registry is the easiest option given the current design.